### DISCUSSION TITLE
Type with relations

### DISCUSSION THREAD

**Discussion by pt-garyblake** [prisma/prisma#19116] [Q&A]
I've got this type coming out OK but without the relations so I am getting type errors

export type TContent = Prisma.contentGetPayload<true>;

model content {
  id                    Int                 @id @default(autoincrement())
  name                  String?             @db.VarChar(255)
  content_items         content_items[]
  content_type_group    content_type_group? @relation(fields: [content_type_group_id], references: [id])
  content_type_group_id Int?
}

It's nagging that the content_type_group_id is not type of content.

Can someone please point me in the right direction?

**Comment by nurul3101** [comment]
Hello @pt-garyblake 👋 

By default, the relation fields are not included in the generated types, you would need to manually add them through Prisma.validator API.
Here's a guide on how to achieve it: [Reference](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type)

**Reply by nurul3101** [reply]
Do let us know in case you need any assistance.

**Reply by pt-garyblake** [reply]
I can see how I can create the objects from the validator but using typeof for each of them seems cumbersome.

Is there something I have missed?

**Reply by nurul3101** [reply]
Is there any particular reason you feel using `typeof` with `Prisma.validator` cumbersome? I believe it would be better than defining types manually.

I can pass the feedback to the team so that we can improve.

**Reply by nurul3101** [reply]
Please feel free to share any feedback you might have.

**Reply by nurul3101** [reply]
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by pt-garyblake** [reply]
I've not been able to get the solution working, I am unsure if it's my fault but I don't have the time to investigate this at the moment.

**Reply by nurul3101** [reply]
Thanks for getting back and letting us know. Please feel free to reopen this GitHub Discussion if needed.
